AxonDeepSeg/segment.py

```diff
@@ -143,6 +143,24 @@ def segment_images(
 
     path_images = [convert_path(p) for p in path_images]
     path_model = convert_path(path_model)
+
+    available_models = ads.get_existing_models_list()
+    if available_models is None:
+        available_models = []
+    if path_model.stem not in available_models:
+            try:
+                print('Model not found, attempting to download')
+                # Call download models from the AxonDeepSeg/download_model.py module
+
+                import AxonDeepSeg.download_model as download_model
+                download_model.download_model()
+
+                available_models = ads.get_existing_models_list()
+                assert path_model.stem in available_models
+            except:
+                print('Could not download models, try again.')
+                quit()
```

Comment:
Might be good to report the full exception message to help when debugging issues with users?

```suggestion
            except Exception as e:
                raise Exception('Could not download models, try again.') from e
```

Result would look something like this:

```
Traceback (most recent call last):
  File "segment.py", line 159, in <module>
    assert path_model.stem in available_models
AssertionError: path_model.stem not in available models

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "segment.py", line 4, in <module>
    raise Exception('Could not download models, try again.') from e
Exception: Could not download models, try again.
```

If this is too complicated, then there are [cleaner options](https://stackoverflow.com/a/75549200), but I would also understand just keeping things simple and only printing a simple message.

Reply:
I'm also curious about the use of `quit()` here, given that usually it's [not recommended](https://www.geeksforgeeks.org/python-exit-commands-quit-exit-sys-exit-and-os-_exit/).

Is it necessary for Napari, maybe?